apollo-router/src/configuration/metrics.rs

```diff
@@ -1,150 +1,139 @@
+            // If the value is not set we don't specify the attribute.
+            None => {
+                attributes.insert(attr_name, false.into());
             }
-        });
 
-        MetricsHandle { _guard: guard }
+            _ => {}
+        };
     }
-}
 
-impl Metrics {
-    pub(crate) fn log_usage_metrics(&mut self) {
-        // We have to have a macro here because tracing requires it. However, we also need to cache the metrics as json path is slow.
+    pub(crate) fn populate_config_instruments(&mut self, yaml: &serde_json::Value) {
         // This macro will query the config json for a primary metric and optionally metric attributes.
-        // The results will be cached for the next iteration.
 
         // The reason we use jsonpath_rust is that jsonpath_lib has correctness issues and looks abandoned.
         // We should consider converting the rest of the codebase to use jsonpath_rust.
 
         // Example usage:
-        // log_usage_metrics!(
+        // populate_usage_instrument!(
         //             value.apollo.router.config.authorization, // The metric name
         //             "$.authorization", // The path into the config
         //             opt.require_authentication, // The name of the attribute
         //             "$[?(@.require_authentication == true)]" // The path for the attribute relative to the metric
         //         );
 
-        macro_rules! log_usage_metrics {
+        macro_rules! populate_config_instrument {
             ($($metric:ident).+, $path:literal) => {
-                let metric_name = stringify!($($metric).+).to_string();
-                let metric = self.metrics.entry(metric_name.clone()).or_insert_with(|| {
-                    if JsonPathInst::from_str($path).expect("json path must be valid").find_slice(&self.yaml).first().is_some() {
+                let instrument_name = stringify!($($metric).+).to_string();
+                self.data.entry(instrument_name.clone()).or_insert_with(|| {
+                    if JsonPathInst::from_str($path).expect("json path must be valid").find_slice(yaml).first().is_some() {
                         (1, HashMap::new())
                     }
                     else {
                         (0, HashMap::new())
                     }
                 });
-
-                // Now log the metric
-                tracing::info!($($metric).+ = metric.0);
-
             };
             ($($metric:ident).+, $path:literal, $($($attr:ident).+, $attr_path:literal),+) => {
-                let metric_name = stringify!($($metric).+).to_string();
-                let metric = self.metrics.entry(metric_name.clone()).or_insert_with(|| {
-                    if let Some(value) = JsonPathInst::from_str($path).expect("json path must be valid").find_slice(&self.yaml).first() {
+                let instrument_name = stringify!($($metric).+).to_string();
+                self.data.entry(instrument_name.clone()).or_insert_with(|| {
```

Comment:
is this clone necessary ?

Reply:
[a4e3605](https://github.com/apollographql/router/pull/4364/commits/a4e36056787230e0182223ffb78d9126e362dca1)